From 9995c5c234ece6888544cdbe6578d47e83dea0b5 Mon Sep 17 00:00:00 2001
From: Arnout Engelen <arnout@bzzt.net>
Date: Sat, 13 Apr 2024 12:54:28 +0200
Subject: [PATCH] workaround for reproducibility issue

Workaround for the issues described in
https://bugzilla.samba.org/show_bug.cgi?id=15632

Probably `$n` should not have been a hash in the first place,
but this at least seems like an incremental improvement and
makes the package reproducible again.
---
 pidl/lib/Parse/Pidl/Typelist.pm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pidl/lib/Parse/Pidl/Typelist.pm b/pidl/lib/Parse/Pidl/Typelist.pm
index 31ea19e357c..f66e8de5fc7 100644
--- a/pidl/lib/Parse/Pidl/Typelist.pm
+++ b/pidl/lib/Parse/Pidl/Typelist.pm
@@ -315,7 +315,8 @@ sub mapType($$)
 	return mapType($t->{DATA}, $n) if ($t->{TYPE} eq "TYPEDEF");
 	return mapScalarType($n) if ($t->{TYPE} eq "SCALAR");
 	return "enum $n" if ($t->{TYPE} eq "ENUM");
-	return "struct $n" if ($t->{TYPE} eq "STRUCT" or $t->{TYPE} eq "INTERFACE");
+	return "struct $n" if (($t->{TYPE} eq "STRUCT" or $t->{TYPE} eq "INTERFACE") and not ref($n) eq "HASH");
+	return "struct $n->{NAME}" if ($t->{TYPE} eq "STRUCT" or $t->{TYPE} eq "INTERFACE");
 	return "union $n" if ($t->{TYPE} eq "UNION");
 	return mapScalarType(bitmap_type_fn($t)) if ($t->{TYPE} eq "BITMAP");
 	return "struct $n" if ($t->{TYPE} eq "PIPE");
-- 
GitLab

